File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/dto/GetListedStockPricesResponse.java
package com.kbank.convenience.stock.stock.mediation.client.dto;

import lombok.Builder;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Builder
public record GetListedStockPricesResponse(
    Long maxPrice,
    Long minPrice,
    Double meanPrice,
    List<GetListedStockPricesSubResponse> list
) {
    @Builder
    public record GetListedStockPricesSubResponse(
            LocalDate baseDate,
            String itemCodeNumber,
            String stockKoreanName,
            Long openPrice,
            Long highPrice,
            Long lowPrice,
            Long closePrice,
            Long volume,
            Long value,
            Long changePrice,
            Double changeRate,
            Long alreadyIssuedStock,
            Long marketPriceTotal,
            String tradingStop
    ) {

    }
}

File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/dto/GetListedStockPricesResponse.kt
package com.kbank.convenience.stock.stock.mediation.client.dto

import lombok.Builder
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate

@Builder
@JvmRecord
data class GetListedStockPricesResponse(val maxPrice: Long,
                                        val minPrice: Long,
                                        val meanPrice: Double,
                                        val list: List<GetListedStockPricesSubResponse>
) {
    @Builder
    @JvmRecord
    data class GetListedStockPricesSubResponse(
                                          val baseDate: LocalDate,
                                          val openPrice: Long,
                                          val highPrice: Long,
                                          val lowPrice: Long,
                                          val closePrice: Long,
                                          val volume: Long,
                                          val value: Long,
                                          val changePrice: Long,
                                          val changeRate: Double,
                                          val alreadyIssuedStock: Long,
                                          val marketPriceTotal: Long,
                                          val tradingStop: String
    )
}
